pkg/deploy/generator/scripts/rpVMSS.sh

```diff
@@ -581,7 +581,11 @@ if [ -f \$NEW_CERT_FILE ]; then
   else
     sed -i -ne '1,/END CERTIFICATE/ p' \$NEW_CERT_FILE
   fi
-  if ! diff $NEW_CERT_FILE $CURRENT_CERT_FILE >/dev/null 2>&1; then
+
+  NEW_CERT_SN=$(openssl x509 -in $NEW_CERT_FILE -noout -serial | awk -F= '{print $2}')
+  CURRENT_CERT_SN=$(openssl x509 -in $CURRENT_CERT_FILE -noout -serial | awk -F= '{print $2}')
+  if [ "$NEW_CERT_SN" != "$CURRENT_CERT_SN" ]; then
+    echo updating certificate for \$COMPONENT
```

Comment:
What's the purpose of escaping `COMPONENT` here? I think we want to display the variable's value?

Reply:
Also if you make use of `test` with double square brackets, you don't have to quote the left variable.  
The right is still expanded, has some good use cases.  
```bash
if [[ $NEW_CERT_SN != "$CURRENT_CERT_SN" ]]; then
    do stuff
fi
```

Reply:
Because it is being added inline with cat > filename <<EOF .... and we want the $ sign to be included.  Which points out I missed a few escapes.

Reply:
Makes sense, I don't use `cat > file <<EOF` often so it looked out of place.